File: practice/greedy/change/Change.java
package greedy.change;

import java.util.Arrays;
import java.util.List;

class Change {

    private static final Integer FIVE_HUNDRED_COIN = 500;
    private static final Integer HUNDRED_COIN = 100;
    private static final Integer FIFTY_COIN = 50;
    private static final Integer TEN_COIN = 10;

    public static void main(String[] args){
        int numberOfCoin = 0;
        int N = 3500;

        List<Integer> coins = Arrays.asList(FIVE_HUNDRED_COIN, HUNDRED_COIN, FIFTY_COIN, TEN_COIN);

        for (Integer coin : coins) {
            numberOfCoin += (N / coin);
            N %= coin;

            if (N == 0) {
                break;
            }
        }

        System.out.println(numberOfCoin);
    }
}

File: practice/greedy/change/change.py
'''
500, 100, 10원
N원일 때 거슬러줘야하는 동전의 최소 갯수
N은 항상 10의 배수

N = 1260
'''
coins = [500,100,10]

N = 1260
cnt = 0

for coin in coins:
    cnt += N//coin
    N %= coin

print(cnt)